framework/mongo/utils.py

```diff
@@ -63,10 +63,18 @@ def wrapper(cls):
 
 def get_or_http_error(Model, pk):
     instance = Model.load(pk)
+
     if getattr(instance, 'is_deleted', False):
-        raise HTTPError(http.GONE, data=dict(
-            message_long="This resource has been deleted"
-        ))
+        if hasattr(instance, 'registrations'):
```

Comment:
It's not necessary, it was just meant address the the original complaint and nothing else. The issue was so minor I didn't see this message would obviously be important in non-registration related contexts. So I took it out.